backend/src/routes/index.ts

```diff
@@ -1,14 +1,21 @@
+import authRouter from './auth';
 import classroomsRouter from './classroom';
-import healthchecksRouter from './healthcheck';
+import healthcheckRouter from './healthcheck';
+import passport from 'passport';
 import { Router } from 'express';
 import teachersRouter from './teacher';
 import usersRouter from './user';
 
 const router = Router();
+const protectedRouter = Router();
 
-router.use('/classrooms', classroomsRouter);
-router.use('/healthchecks', healthchecksRouter);
-router.use('/teachers', teachersRouter);
-router.use('/users', usersRouter);
+router.use('/auth', authRouter);
+router.use('/healthcheck', healthcheckRouter);
+
+protectedRouter.use('/classrooms', classroomsRouter);
+protectedRouter.use('/teachers', teachersRouter);
+protectedRouter.use('/users', usersRouter);
+
+router.use('/', passport.authenticate('jwt', { session: false }), protectedRouter);
```

Comment:
I believe this part can be written in a more straightforward way like so:

```js
const router = Router();
router.use('/auth', authRouter);
router.use('/healthcheck', healthcheckRouter);

router.use(passport.authenticate('jwt', { session: false });

router.use('/classrooms', classroomsRouter);
router.use('/teachers', teachersRouter);
router.use('/users', usersRouter);
```
This should be valid code since we are just adding middleware before all of the protected routes while using the same router.